File: ukol-5-priklady/program.py
import random

class World:
    def __init__(self, lights, start_pos=0):
        self.lights = list(lights)
        self.pos = start_pos % len(self.lights)

    def isLightOn(self):
        return self.lights[self.pos]

    def toggleLight(self):
        self.lights[self.pos] = not self.lights[self.pos]

    def moveNext(self):
        self.pos = (self.pos + 1) % len(self.lights)

    def movePrev(self):
        self.pos = (self.pos - 1) % len(self.lights)

world: World

def isLightOn() -> bool:
    return world.isLightOn()


def toggleLight() -> None:
    world.toggleLight()


def moveNext() -> None:
    world.moveNext()


def movePrev() -> None:
    world.movePrev()


def countRooms() -> int:
    """
    TODO: Implementovat
    """
    return 0


def run_test(n: int, lights=None, start_pos: int = 0):
    global world
    world = World(lights, start_pos)

    result = countRooms()
    ok = (result == n)
    print(f"Test n={n}, start={start_pos}, lights={lights} -> "
          f"countRooms()={result}  [{'OK' if ok else 'FAIL'}]")
    return ok


if __name__ == "__main__":
    all_ok = True
    all_ok &= run_test(3, [True, False, True], 0)
    all_ok &= run_test(5, [False, True, True, False, True], 2)
    # Přidej další testy

    if not all_ok:
        print("Některé testy selhaly – opravte countRooms().")
    else:
        print("Všechny testy prošly.")

File: ukol-5-priklady/Program.cs
using System;
using System.Collections.Generic;

public static class Program
{
    private static World world;

    public static bool isLightOn() => world.IsLightOn();
    public static void toggleLight() => world.ToggleLight();
    public static void moveNext() => world.MoveNext();
    public static void movePrev() => world.MovePrev();

    public static int countRooms()
    {
        // TODO: Implementovat
        return 0;
    }

    private static bool RunTest(int n, bool[] lights, int startPos)
    {
        world = new World(lights, startPos);
        int result = countRooms();
        bool ok = result == n;

        Console.WriteLine($"Test n={n}, start={startPos}, lights=[{string.Join(",", lights)}] " +
                          $"=> countRooms()={result}  {(ok ? "OK" : "FAIL")}");
        return ok;
    }

    public static void Main()
    {
        bool allOk = true;
        allOk &= RunTest(3, new[] { true, false, true }, 0);
        allOk &= RunTest(5, new[] { false, true, true, false, true }, 2);
        // Přidej další testy

        if (!allOk)
            Console.WriteLine("Některé testy selhaly – opravte countRooms().");
        else
            Console.WriteLine("Všechny testy prošly.");
    }
}

public class World
{
    private List<bool> lights;
    private int pos;

    public World(bool[] initialLights, int startPos)
    {
        lights = new List<bool>(initialLights);
        pos = ((startPos % lights.Count) + lights.Count) % lights.Count;
    }

    public bool IsLightOn() => lights[pos];

    public void ToggleLight() => lights[pos] = !lights[pos];

    public void MoveNext() => pos = (pos + 1) % lights.Count;

    public void MovePrev() => pos = (pos - 1 + lights.Count) % lights.Count;
}
